src/lib.rs

```diff
@@ -307,6 +307,31 @@ impl Map {
         }
         maximum_ts
     }
+
+    /// Computes the rectangle on the image where the sprite is stored for the given tile ID.
+    /// If the ID is not found in any tileset, or if there is no image associated with the tileset, `None` is returned.
+    /// On success, returns `Some(x, y, w, h)`, where `(x, y)` is the coordinates of the top-left corner, and `(w, h)` are the width and height of the rectangle
+    pub fn get_tile_rectangle_by_id(&self, id: u32) -> Option<(u32, u32, u32, u32)> {
+        let tileset = self.get_tileset_by_gid(id)?;
+        let img = tileset.images.get(0)?; // we suppose there is only 1 image per tileset
+
+        let id = id - 1;
+        let columns =
+            (img.width as u32 - 2 * tileset.margin) / (tileset.spacing + tileset.tile_width);
```

Comment:
This is slightly wrong. Since spacing is between tiles, it doesn't apply to one of the columns (first or last depending on how you think about it)
(ex. tileset with image width = 572, margin = 2, spacing = 4 should be 13 columns, but old method gave ~12.91)
```suggestion
            (img.width as u32 - 2 * tileset.margin - tileset.tile_width) / (tileset.spacing + tileset.tile_width) + 1;
```